topic1/src/otherKeyPoints/Cash.java

```diff
@@ -0,0 +1,14 @@
+package otherKeyPoints;
+
+public class Cash implements Pay {
+
+	@Override
+	public void pay(ShoppingCart shoppingCart, int paymentTransactionNumber) {
+
+		System.out.println("This is really a nice Customer :)\"NOW INFORMATION IS NEEDED\" amount ="
+				+ (shoppingCart.getTotalAmount() - (shoppingCart.getTheMostExpensiveItem().getPrice() * 0.9)));
```

Comment:
It's fine, but try to avoid the calculations as parameters... lets move this into another method !